### ISSUE TITLE
Parser fails silently on records with invalid identifiers

### ISSUE THREAD

**Issue by nics** [nickng/bibtex#21]
The parser silently skips this record I exported from Scopus and returns no error:

```
@ARTICLE{Cè2022411,
author={Cè, M.},
title={THE SECONDARY INCIPIT OF THE ODYSSEY (OD. 9.39): QUOTATION, TRANSLATION, AND ADAPTATION IN THE ANCIENT RECEPTION OF HOMER},
journal={Classical Philology},
year={2022},
volume={117},
number={3},
pages={411-437},
doi={10.1086/720155},
note={cited By 0},
url={https://www.scopus.com/inward/record.uri?eid=2-s2.0-85133505893&doi=10.1086%2f720155&partnerID=40&md5=11af9d3ef7a64fc7db44afa582afb7b6},
affiliation={Bayerische Akademie der Wissenschaften, Germany},
abstract={Although recent scholarship has studied the incipit as a privileged intertextual locus in Latin poetry, the comparable role of work-internal beginnings in classical literature has been largely overlooked. This article argues that the opening sentence of Odysseus’ Apologue functions as the Odyssey’s secondary beginning from Odysseus’ point of view—a role that ancient audiences of the poem recognized and developed. Specifically, I demonstrate that later Greek and Latin authors use the Odyssey’s secondary incipit to emphasize narratorial subjectivity and to mark new literary beginnings. Consequently, my research calls for a broader reconsideration of secondary incipits in Graeco-Roman literature. © 2022 The University of Chicago. All rights reserved.},
publisher={University of Chicago Press},
issn={0009837X},
language={English},
abbrev_source_title={Cl. Philol.},
document_type={Article},
source={Scopus},
}

```

**Comment by nickng** [comment]
I got:
```
parse failed at 1:11: syntax error
```
How did you use the parser?

**Comment by nics** [comment]
Hi @nickng, we're using `bibtex.Parse`.  I made a condensed example at https://go.dev/play/p/mpS3Yc9cklh

```
	bib, err := bibtex.Parse(strings.NewReader(rec))
	if err != nil {
		log.Fatal(err)
	}
```

**Comment by nickng** [comment]
I'm still getting error locally but just remembered I haven't tagged recent changes with recent error-related changes.
I can reproduce this in v1.1.0 (previous latest tagged version) and it's gone in v1.2.0 🆕.
Could you try updating to v1.2.0?

**Comment by nics** [comment]
Thanks a lot, get the error now.

**Comment by nickng** [comment]
Fixed in v1.2.0
